### DISCUSSION TITLE
Regolith 3 - lockscreen swaylock

### DISCUSSION THREAD

**Discussion by eljejer** [regolith-linux/regolith-desktop#903] [Q&A]
Hello
Trying out Sway on Regolith3. So far so GOOD!
According to [this](https://regolith-desktop.com/docs/howtos/change-lockscreen/) howto, it should be possible to change the lock screen program.
It is not going as expected though in a Wayland session (swaylock not invoked) where in an X11 session it does (betterlockscreen invoked).

Xresources:

```
regolith.wallpaper.file: /usr/share/backgrounds/WP_20150405_001.jpg
wm.program.lock: /home/eljejer/scripts/myi3lock.sh
wm.bar.position: top
wm.gaps.inner.size: 1
wm.program.files: /usr/bin/pcmanfm-qt
wm.binding.files: e
!regolith.look.path: /usr/share/regolith-look/nord
regolith.look.path: /usr/share/regolith-look/lascaill
```

my myi3lock.sh:

```
case $WAYLAND_DISPLAY in
	"wayland-1")	
			swaylock
			;;
	*)
			betterlockscreen -l dimblur
			;;
esac
```

**Comment by SoumyaRanjanPatnaik** [comment]
Hi. This is happening due to incorrect resource names in the gtklock config file. Should be fixed in https://github.com/regolith-linux/regolith-wm-config/pull/18
